### PR TITLE
fix(fa): size rescale UB tiles with aligned head stride

### LINKED ISSUE #135: [BUG]: FA-2/3/4 precision issue with non-16-aligned head dimensions (e.g. D=200) caused by PV/rescale UB overflow
### Preflight checklist

- [x] I have searched the existing issues and confirmed this is not a duplicate.
- [x] I can reproduce this on the latest release.
- [x] I have removed any sensitive information from logs or code snippets.

### Environment

- Device (e.g., Ascend 910B): 910
- CANN version: 9.1.0
- PyTorch version: 2.10
- torch_npu version: 2.10
- Python version: 3.11
- flash_attn_npu version: main
- etc.

### Bug Description

On Ascend 910, `test_fa_custom_ops` shows an accuracy issue when `head_dim` is not 16-element aligned.

Minimal reproducer:

- dtype: FP16
- batch size: 1
- query heads: 2
- KV heads: 1
- query sequence length: 64
- KV sequence length: 64
- head dimension: 200
- layout: BSND
- cache mode: non-paged
- causal: false

The QK matmul output has been verified as correct. The issue occurs in the post-softmax PV/output-rescale path.

### Steps to Reproduce

NA

### Error output

```shell
NA
```

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by jianshuqiang-cpu**
## Related Issue

Fixes #135 

## Summary

`rescale_o.hpp` calculates the number of rows per UB tile using the logical head dimension:

## Validation

image

## Additional Information

NA

### CODE REVIEW

**Comment by jianshuqiang-cpu** [comment]
## NPU CI

- 触发人: @jianshuqiang-cpu
- 模式: `quick` (含 Example ST)
- PR: #136  commit: `c6cf13ee`
- 状态: 失败
- Actions: [https://github.com/MinghuasLab/flash-attention-npu/actions/runs/31929076231](https://github.com/MinghuasLab/flash-attention-npu/actions/runs/31929076231)
